### PR TITLE
Add ruby on rails example instrumented using OpenTelemetry 

### PR DESCRIPTION
**Pull request by chtushar**
- Add ruby on rails example instrumented using OpenTelemetry

### CODE REVIEW

File: ruby/rails-api/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+5. Run the Ruby on rails application:
+
+```bash
+bin/rails server
+```
+
+6. Once the server is running, you can access the application at
+   `http://localhost:3000` by default. Where you can make CRUD operations. The
+   API endpoints are:
```
Comment (prathamesh-sonpatki) [inline ruby/rails-api/README.md:38]:
```suggestion
6. Once the server is running, you can access the application at
   `http://localhost:3000` by default. The
   API endpoints are:
```

File: ruby/rails-api/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+4. In the `.env` file, set the value of `OTEL_EXPORTER_OTLP_HEADERS` to the OTLP
+   Authorization Header obtained from the Last9 dashboard and make sure the
+   value of the header is URL encoded.
+
+```bash
+OTEL_EXPORTER_OTLP_HEADERS="Authorization=<BASIC_AUTH_HEADER>"
+```
+
```
Comment (prathamesh-sonpatki) [inline ruby/rails-api/README.md:29]: Step 5 should be how to configured in the `config/initializers/opentelemetry.rb`

```
require 'opentelemetry/sdk'
require 'opentelemetry/exporter/otlp'
require 'opentelemetry/instrumentation/all'

OpenTelemetry::SDK.configure do |c|
  c.service_name = 'ruby-on-rails-api-service'

  config = {'OpenTelemetry::Instrumentation::fs' => { enabled: false }}
  c.use_all(config) # enables all instrumentation!
end
```

**Review by prathamesh-sonpatki — CHANGES_REQUESTED** [review]

File: ruby/rails-api/.env.example.erb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+OTEL_SERVICE_NAME=ruby-on-rails-api-service
+OTEL_EXPORTER_OTLP_ENDPOINT=https://otlp.last9.io
+OTEL_EXPORTER_OTLP_HEADERS=
+OTEL_TRACES_EXPORTER=console,otlp
+OTEL_METRICS_EXPORTER=otlp
```
Comment (prathamesh-sonpatki) [inline ruby/rails-api/.env.example.erb:5]: @chtushar if this option is not used, please remove it.

File: ruby/rails-api/config/initializers/opentelemetry.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+require 'opentelemetry/sdk'
+require 'opentelemetry/exporter/otlp'
+require 'opentelemetry/instrumentation/all'
+
+OpenTelemetry::SDK.configure do |c|
+  c.service_name = 'ruby-on-rails-api-service'
+
+  config = {'OpenTelemetry::Instrumentation::fs' => { enabled: false }}
```
Comment (prathamesh-sonpatki) [inline ruby/rails-api/config/initializers/opentelemetry.rb:8]: @chtushar Have you verified this config is true? I checked [here](https://github.com/open-telemetry/opentelemetry-ruby-contrib/tree/main/instrumentation), there is no `fs` instrumentation for Ruby.

**Review by prathamesh-sonpatki — CHANGES_REQUESTED** [review]

File: ruby/rails-api/README.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+   The Auth header is required in the next step.
+
+3. Next, run the commands below to set the environment variables.
+
+```bash
+touch .env
+cp .env.example .env
+```
+
+4. In the `.env` file, set the value of `OTEL_EXPORTER_OTLP_HEADERS` to the OTLP
```
[the remark this answers is not in the document]
Comment (chtushar) [reply ruby/rails-api/README.md:22]: Screenshot 2024-07-21 at 10 36 32 PM

Not copy pasted. It is duly checked. It works with the encoding. The traces in "traces.png" are captured after following the instructions in the README.md file.

File: ruby/rails-api/README.md (hunk 2 above)
Comment (chtushar) [reply ruby/rails-api/README.md:29]: We are doing that in the docs. The README.md is a guide to set up this example.

File: ruby/rails-api/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (prathamesh-sonpatki) [reply ruby/rails-api/README.md:22]: @chtushar Does it work without encoding?

[the remark this answers is not in the document]
Comment (prathamesh-sonpatki) [reply ruby/rails-api/README.md:22]: @chtushar Please respond to this.

**Review by prathamesh-sonpatki — APPROVED** [review]

**Pull request merged**